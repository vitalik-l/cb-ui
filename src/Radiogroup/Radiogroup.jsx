import React, {Component} from 'react';
import PropTypes from 'prop-types';
import omit from '../utils/omit';

class Radiogroup extends Component {
    static propTypes = {
        value: PropTypes.any,
        children: PropTypes.any,
        className: PropTypes.string,
        onChange: PropTypes.func
    };

    onClick = e => {
        const clickedValue = e.currentTarget.getAttribute('data-value');
        if (clickedValue == null) return;
        this.props.onChange && this.props.onChange(clickedValue, e);
    };

    render() {
        const {children, value, className} = this.props;
        const childrenItems = React.Children.map(children, (child) => {
            let childProps = {};
            if (value && !child.props.hasOwnProperty('active')) childProps.active = child.props.value == value;
            if (this.props.onChange && !child.props.onClick) childProps.onClick = this.onClick;
            return React.cloneElement(child, childProps);
        });
        return (
            <ul className={className || 'cb-radiogroup'} {...omit(this.props, ['children', 'value', 'onChange'])}>
                {childrenItems}
            </ul>
        )
    }
}

export default Radiogroup;