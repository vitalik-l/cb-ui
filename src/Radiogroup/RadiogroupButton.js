import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import omit from '../utils/omit';

export default class RadiogroupButton extends React.PureComponent {
    render() {
        const {children, active, value, className} = this.props;
        return (
            <li className={classNames(className, {active})} data-value={value} {...omit(this.props, ['className', 'value', 'children', 'active'])}>
                {children}
            </li>
        );
    }
}

RadiogroupButton.propTypes = {
    children: PropTypes.any,
    active: PropTypes.bool,
    value: PropTypes.any
};