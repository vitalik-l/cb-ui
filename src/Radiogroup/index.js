import Radiogroup from './Radiogroup';
import RadiogroupButton from './RadiogroupButton';

export default Radiogroup;
export {RadiogroupButton};