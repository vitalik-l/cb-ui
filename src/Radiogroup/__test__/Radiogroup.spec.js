import React from 'react';
import {shallow} from 'enzyme';
import Radiogroup from '../Radiogroup';
import RadiogroupButton from '../RadiogroupButton';

describe('Radiogroup', () => {
    const wrapper = shallow(
        <Radiogroup value={1}>
            <RadiogroupButton value={1}>1</RadiogroupButton>
            <RadiogroupButton value={2}>2</RadiogroupButton>
            <RadiogroupButton value={3}>3</RadiogroupButton>
        </Radiogroup>
    );

    it('default value is 1. Should set the "active" class to the RadiogroupButton with the value prop equal to 1', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('change value to 2. Should set the "active" class to the RadiogroupButton with the value prop equal to 2', () => {
        wrapper.setProps({value: 2});
        expect(wrapper).toMatchSnapshot();
    });
});