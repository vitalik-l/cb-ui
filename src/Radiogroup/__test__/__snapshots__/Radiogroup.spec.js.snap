// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radiogroup change value to 2. Should set the "active" class to the RadiogroupButton with the value prop equal to 2 1`] = `
<ul
  className="cb-radiogroup"
>
  <RadiogroupButton
    active={false}
    value={1}
  >
    1
  </RadiogroupButton>
  <RadiogroupButton
    active={true}
    value={2}
  >
    2
  </RadiogroupButton>
  <RadiogroupButton
    active={false}
    value={3}
  >
    3
  </RadiogroupButton>
</ul>
`;

exports[`Radiogroup default value is 1. Should set the "active" class to the RadiogroupButton with the value prop equal to 1 1`] = `
<ul
  className="cb-radiogroup"
>
  <RadiogroupButton
    active={true}
    value={1}
  >
    1
  </RadiogroupButton>
  <RadiogroupButton
    active={false}
    value={2}
  >
    2
  </RadiogroupButton>
  <RadiogroupButton
    active={false}
    value={3}
  >
    3
  </RadiogroupButton>
</ul>
`;
